analyzers/tests/SonarAnalyzer.Test/TestCases/VariableShadowsField.cs

```diff
@@ -159,4 +159,41 @@ public void OutParameter(out object parameter)
             parameter = null;
         }
     }
+
+    public class Base
+    {
+        private int privateInstanceField = 1;
+        private int privateInstanceProperty { get; set; } = 1;
+        private static int privateStaticField = 1;
+        private static int privateStaticProperty = 1;
+
+        protected int protectedInstanceField = 1;
+        protected int protectedInstanceProperty { get; set; } = 1;
+        protected static int protectedStaticField = 1;
+        protected static int protectedStaticProperty = 1;
+
+        public static void BaseStaticMethod()
+        {
+            var privateInstanceField = 2;       // Compliant (instance field is not accessible from static method)
+            var privateInstanceProperty = 2;    // Compliant (instance property is not accessible from static method)
+
+            var privateStaticField = 2;         // Noncompliant
+            var privateStaticProperty = 2;      // Noncompliant
+        }
```

Comment:
I would add the instance counterpart of this method, just to document that it works as intended.

```csharp

        public void BaseInstanceMethod()
        {
            var privateInstanceField = 2;       // Noncompliant
            var privateInstanceProperty = 2;    // Noncompliant

            var privateStaticField = 2;         // Noncompliant
            var privateStaticProperty = 2;      // Noncompliant
        }
```